test/units/parsing/test_splitter.py

```diff
@@ -173,3 +177,9 @@ def test_split_args_error(args):
 def test_parse_kv_error(args):
     with pytest.raises(AnsibleParserError):
         parse_kv(args)
+
+
+@pytest.mark.parametrize("args", REMAINDER_ERROR)
+def test_parse_kv_remainder_error(args):
+    with pytest.raises(ValueError):
```

Comment:
Verify that the `ValueError` encountered is the correct one, and not due to an unexpected failure:

```suggestion
    with pytest.raises(ValueError, match='Invalid k=v detected'):
```